common/status-effects/invisibility-potion.ts

```diff
@@ -21,10 +21,16 @@ export const InvisibilityPotionHeadsEffect: StatusEffect<PlayerComponent> = {
 	) {
 		observer.subscribeWithPriority(
 			player.opponentPlayer.hooks.beforeAttack,
-			beforeAttack.EFFECT_MODIFY_DAMAGE,
+			beforeAttack.MODIFY_DAMAGE,
 			(attack) => {
 				if (!attack.isType('primary', 'secondary')) return
 				attack.multiplyDamage(effect.entity, 0)
+			},
+		)
+		observer.subscribeWithPriority(
+			player.opponentPlayer.hooks.onTurnEnd,
+			onTurnEnd.ON_STATUS_EFFECT_TIMEOUT,
+			() => {
 				effect.remove()
```

Comment:
Invisibility status effects should be persistent until they modify an attack
```suggestion
				if (multipliedDamage) effect.remove()
```

Reply:
Good catch! I didn't realize it should last multiple turns.